src/circles.rs

```diff
@@ -1,121 +1,131 @@
+#[cfg(feature = "vec")]
 use svg::node::element;
-use hex;
 
+#[cfg(feature = "pix")]
 const BACKGROUND_COLOR: [u8; 4] = [255, 255, 255, 0];
-pub const FOREGROUND_COLOR: [u8; 4] = [238, 238, 238, 255];
+pub(crate) const FOREGROUND_COLOR: [u8; 4] = [238, 238, 238, 255];
 
-/// Struct to store information about the circle
+/// Information about the circle
+#[cfg(feature = "pix")]
 struct Circle {
-    x_center: i32,
-    y_center: i32,
-    radius: i32,
+    x_center: f32,
+    y_center: f32,
+    radius: f32,
     rgba_color: [u8; 4],
 }
 
-
 /// Function to determine if the point (x, y) is within the circle
-fn in_circle (x: i32, y: i32, circle: &Circle) -> bool {
-    (x-circle.x_center).pow(2) + (y-circle.y_center).pow(2) < circle.radius.pow(2)
+#[cfg(feature = "pix")]
+fn in_circle(x: i32, y: i32, circle: &Circle) -> bool {
+    (x as f32 - circle.x_center).powi(2) + (y as f32 - circle.y_center).powi(2)
```

Comment:
nit: readability improves to move casts out of the equation and adding an addition `let x = x as f32` binding. Note that this case is simple enough to leave as is.